runtime/gc_vlhgc/IncrementalGenerationalGC.cpp

```diff
@@ -1987,6 +2015,41 @@ MM_IncrementalGenerationalGC::postConcurrentUpdateStatsAndReport(MM_EnvironmentB
 	env->_cycleState = NULL;
 }
 
+void
+MM_IncrementalGenerationalGC::calculateConcurrentMarkWorkTime(MM_EnvironmentBase *env, MM_ConcurrentPhaseStatsBase *stats)
+{
+	/* Get an estimate of total process time now that concurrent mark increment has finished*/
+	omrthread_process_time_t processEnd;
+	omrthread_get_process_times(&processEnd);
+	U_64 processEndTime = processEnd._systemTime + processEnd._userTime;
+
+	/* Calculate how much process time has elapsed since the start of the concurrent mark increment */
+	U_64 concurrentElapsedTime = processEndTime - stats->_concurrentMarkStartTime;
+
+	/* Work time we attribute to concurrent work is difference of process time of gc threads now vs when concurrent phase started */
+	MM_MarkVLHGCStats markStats = _persistentGlobalMarkPhaseState._vlhgcIncrementStats._markStats;
+	double concurrentGCRatio = 0.5;
+
+	if (markStats._concurrentGCThreadsEndTimeSum != markStats._concurrentGCThreadsStartTimeSum) {
+		/* If the platform supports getting time for each thread, calculate what % of time gc threads were active relative to application threads. */
+		/* Determine the ratio of time spent doing GC work vs non-gc related work during the concurrent phase */
+		U_64 concurrentGCWorkTime = markStats._concurrentGCThreadsEndTimeSum - markStats._concurrentGCThreadsStartTimeSum;
+		concurrentGCRatio = (double)concurrentGCWorkTime/concurrentElapsedTime;
+
+		/* If there was a clock error, concurrentGCWorkTime might be too high or negative. Make sure ratio is between 0.1 and 0.8 */
+		concurrentGCRatio = OMR_MIN(concurrentGCRatio, 0.8);
+		concurrentGCRatio = OMR_MAX(concurrentGCRatio, 0.1);
```

Comment:
Nice safety measures! Why not 0.9 (to be symmetric with 0.1)?
BTW, I'd like to reduce GC thread count in GMP increments, but it should be a separate change and experimentation.

Reply:
No reason in particular. I can change to symmetrical values if you prefer, 0.1/0.9 or 0.2/0.8, let me know

Reply:
ok, let's flip a coin and say 0.1/0.9